src/v/cluster/partition_leaders_table.cc

```diff
@@ -118,17 +119,17 @@ void partition_leaders_table::update_partition_leader(
         return;
     }
 
-    auto it = _leaders.find(key);
-    if (it == _leaders.end()) {
-        auto [new_it, _] = _leaders.emplace(
-          leader_key{
-            model::topic_namespace(ntp.ns, ntp.tp.topic), ntp.tp.partition},
-          leader_meta{
-            .current_leader = leader_id,
-            .update_term = term,
-            .partition_revision = revision_id});
-        it = new_it;
-    } else {
+    auto [t_it, _] = _topic_leaders.emplace(
+      model::topic_namespace_view(ntp), partition_leaders{});
+
+    auto [p_it, new_entry] = t_it->second.emplace(
+      ntp.tp.partition,
+      leader_meta{
+        .current_leader = leader_id,
+        .update_term = term,
+        .partition_revision = revision_id});
+
+    if (likely(!new_entry)) {
```

Comment:
Should we be using `[[likely]]` and `[[unlikely]]` from C++ instead of the macros now?